src/pivot-table/components/helpers/get-scrollable-areas-dimensions.ts

```diff
@@ -0,0 +1,126 @@
+import type { Rect } from "../../../types/types";
+import { GRID_BORDER } from "../../constants";
+
+type ContainerDims = {
+  scrollable: Rect;
+  fullSize: Rect;
+  sticky: Rect;
+};
+
+interface GetScrollableAreasDimensionsProps {
+  tableRect: Rect;
+
+  containerHeight: number;
+  leftGridHeight: number;
+  topGridHeight: number;
+  dataGridHeight: number;
+  allRowsVisible: boolean;
+
+  totalWidth: number;
+  leftGridWidth: number;
+  rightGridWidth: number;
+  verticalScrollbarWidth: number;
+  horizontalScrollbarHeight: number;
+}
+
+interface GetScrollableAreasDimensionsResult {
+  ROOT_WRAPPER: ContainerDims;
+  LEFT_WRAPPER: {
+    containers: ContainerDims;
+    headerGrid: Rect;
+    leftGrid: Rect;
+  };
+  RIGHT_WRAPPER: {
+    containers: ContainerDims;
+    topGrid: Rect;
+    dataGrid: Rect;
+  };
+}
+
+const getScrollableAreasDimensions = ({
+  tableRect,
+
+  containerHeight,
+  leftGridHeight,
+  topGridHeight,
+  dataGridHeight,
+  allRowsVisible,
+
+  totalWidth,
+  leftGridWidth,
+  rightGridWidth,
+  verticalScrollbarWidth,
+  horizontalScrollbarHeight,
+}: GetScrollableAreasDimensionsProps): GetScrollableAreasDimensionsResult => {
+  const modifiedVerticalScrollbarWidth = verticalScrollbarWidth * (allRowsVisible ? 0 : 1);
+  const modifiedHorizontalScrollbarHeight = horizontalScrollbarHeight * (allRowsVisible ? 0 : 1);
+
+  const childWrappersScrollableContainerHeight = allRowsVisible
+    ? topGridHeight + dataGridHeight + GRID_BORDER + horizontalScrollbarHeight
+    : tableRect.height;
+  const childWrappersStickyContainerHeight = tableRect.height - modifiedHorizontalScrollbarHeight;
+
+  return {
+    ROOT_WRAPPER: {
```

Comment:
All caps tend to indicate that it's a static value/constant. But this is clearly not, so consider renaming it to `rootWrapper`